### PR TITLE
Batch printing

### PR DESCRIPTION
**Pull request by emjay0921**

### CODE REVIEW

File: spp_idqueue/models/id_batch.py [hunk 1]
```diff
@@ -25,3 +25,22 @@ class OpenSPPPrintBatch(models.Model):
...
     id_pdf = fields.Binary("ID PASS")
     id_pdf_filename = fields.Char("ID File Name")
+
+    def generate_batch(self):
+        for rec in self:
+            for queue in rec.queued_ids:
+                queue.on_generate()
+            if not rec.queued_ids.filtered(lambda x: x.status not in ["generated"]):
+                rec.status = "generated"
+                rec.pass_api_param()
```
Comment (jeremi) [inline spp_idqueue/models/id_batch.py]: You should just call the method `rec.queued_ids.generate_cards()`

File: spp_idqueue/wizard/batch_print_wizard.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            batch_name = rec.name
+            if batches_count > 1:
+                batch_name = _("%s - 1", rec.name)
+            batch_id = self.env["spp.print.queue.batch"].create({"name": batch_name})
+            current_batch_count = 1
+            for queue in rec.queue_ids:
+                if not current_batch_count > batches_count:
+                    id_count += 1
+                    if not id_count > rec.max_id_per_batch:
+                        queue.write({"batch_id": batch_id.id})
```
Comment (jeremi) [inline spp_idqueue/wizard/batch_print_wizard.py]: Can we call write only once per batch?

File: spp_idqueue/wizard/batch_print_wizard.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                    else:
+                        if not current_batch_count == batches_count:
+                            batch_name = f"{rec.name or ''} - {current_batch_count + 1}"
+                            batch_id = self.env["spp.print.queue.batch"].create(
+                                {"name": batch_name}
+                            )
+                            queue.write({"batch_id": batch_id.id})
```
Comment (jeremi) [inline spp_idqueue/wizard/batch_print_wizard.py]: call write only once per batch

File: spp_idqueue/wizard/batch_print_wizard.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                res["id_count"] = len(queue_ids)
+            else:
+                raise UserError(_("There are no valid id requests!"))
+            return res
+        else:
+            raise UserError(_("There are no selected id requests!"))
+
+    name = fields.Char(string="Batch Name")
+    id_count = fields.Integer(string="ID Count")
+    max_id_per_batch = fields.Integer(string="Max ID per batch", default=1)
```
Comment (jeremi) [inline spp_idqueue/wizard/batch_print_wizard.py]: let's make the default 20

File: spp_idqueue/wizard/batch_print_wizard.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                        </field>
+                    </group>
+                    <group>
+                        <field name="id_count" readonly="1" />
+                        <field name="max_id_per_batch" />
+                        <field name="batches_count" />
+                    </group>
+                </sheet>
+                <footer>
+                    <button name="print_batch" string="Print Batch" type="object" class="oe_highlight" />
```
Comment (jeremi) [inline spp_idqueue/wizard/batch_print_wizard.xml]: This should be only available once the batch has been generated.

**Review by jeremi — CHANGES_REQUESTED** [review]
Where is the code that call the batch API of ID PASS?

File: spp_idqueue/models/id_batch.py (hunk 1 above)
Comment (emjay0921) [reply spp_idqueue/models/id_batch.py]: Done

File: spp_idqueue/wizard/batch_print_wizard.py (hunk 2 above)
Comment (emjay0921) [reply spp_idqueue/wizard/batch_print_wizard.py]: Done

File: spp_idqueue/wizard/batch_print_wizard.py (hunk 3 above)
Comment (emjay0921) [reply spp_idqueue/wizard/batch_print_wizard.py]: Done

File: spp_idqueue/wizard/batch_print_wizard.py (hunk 4 above)
Comment (emjay0921) [reply spp_idqueue/wizard/batch_print_wizard.py]: Done

File: spp_idqueue/wizard/batch_print_wizard.xml (hunk 5 above)
Comment (emjay0921) [reply spp_idqueue/wizard/batch_print_wizard.xml]: Done, I renamed this to create_batch to avoid confusion.

**Comment by emjay0921** [comment]
> Where is the code that call the batch API of ID PASS?

I first added it in the PDS-SPP, now I removed it there then transfer it here.

**Review by jeremi — APPROVED** [review]

**Pull request merged**